lightning/src/ln/channelmanager.rs

```diff
@@ -868,17 +868,30 @@ pub struct ChannelDetails {
 	pub unspendable_punishment_reserve: Option<u64>,
 	/// The `user_channel_id` passed in to create_channel, or 0 if the channel was inbound.
 	pub user_channel_id: u64,
+	/// Our total balance.  This is the amount we would get if we close the channel.
+	/// This does not include any pending HTLCs which are not yet fully resolved (and, thus, whose
+	/// balance is not available for inclusion in new outbound HTLCs). This further does not include
+	/// any pending outgoing HTLCs which are awaiting some other resolution to be sent.
+	/// This does not consider any on-chain fees.
+	///
+	/// See also [`ChannelDetails::outbound_capacity_msat`]
+	///
+	/// This value is not exact. Due to various in-flight changes, feerate changes, and our
+	/// conflict-avoidance policy, exactly this amount is not likely to be recoverable on close.
```

Comment:
what do you mean by conflict-avoidance policy in that context ?

Reply:
This is just copied from below, so maybe @TheBlueMatt can comment.

Reply:
Yea, don't think the conflict-avoidance policy (ie refusing to send an HTLC when we're low on balance, even if we technically have enough balance) needs to be mentioned here. I think the above paragraph is largely sufficient here.